Remove CAS attributes from System.Drawing

qmfrederik opened the pull request:
Based on what I understand, CAS is deprecated in pretty much all of .NET and not implemented in Mono.

With all the PRs done to get System.Drawing running on .NET Core 2, I realized we're actually not too far from getting it to work on .NET Core 1.x as well.

This is a first step, happy to discuss.

marek-safar commented:
Please don't remove code in referencesource code, use `#if MONO_FEATURE_CAS ... #endif` instead

qmfrederik commented:
@marek-safar OK, I updated the PR. There were a couple of files in the reference source where the using statement was present but not used - I removed the statement instead of #ifdef'ing it. Let me know if you want me to add it back.

akoeplinger commented on mcs/class/referencesource/System/compmod/system/componentmodel/InvalidEnumArgumentException.cs:
```diff
@@ -10,7 +10,6 @@ namespace System.ComponentModel {
     using System.Diagnostics;
     using System.Globalization;
     using System.Runtime.Serialization;
-    using System.Security.Permissions;
```
this is not unused, HostProtection below is in that namespace
